_posts/2023-08-04-deploying-OIDC-webapp-on-openshift.adoc

```diff
@@ -0,0 +1,161 @@
+
+Navigate to the `Client Scopes` tab for `myclient`. Here you will see that some scope values are under `Assigned Default Client Scopes` field and some others are under `Assigned Optional Client Scopes` field. When client scopes are added to `Default`, the claims associated with them are automatically added to the access token you receive without the need to configure them. Changing it to `Optional` ensures that they are not included by default and you will only get access to claims if you configure the corresponding scope values. This list of scope values you see here are the values that RH-SSO allows and it varies for different OpenID providers. RH-SSO also allows you to create new scopes, but we will not be doing that for this blog. 
+
+Now, click on `Roles` and create two roles, `user` and `admin`.
+
+Finally, create a user called `alice` whose `First Name` is Alice and `Last Name` is Smith and assign her the `user` and `admin` roles. Next, navigate to the `Credentials` tab and create a password for Alice. Toggle the `Temporary` switch off so you are not prompted to update the password after the first login. 
+
+== Building Container Image for WildFly
+
+In order to deploy our application on OpenShift, we need to first create a container image of the application using Docker or Podman. In this post we are using Docker, but you can also use Podman using the `podman` keyword, or using an alias.  
+
+Navigate to the `WildFly` project and create a `Dockerfile` inside the root directory of the project. We will be creating an image of the current version of WildFly using the `wildfly-runtime-jdk` image on quay. Quay is a repository for building and storing container images of your projects. 
+
+Head over to https://quay.io/repository/[quay.io] and create a repositories called `wildfly`. For more information on how to create one, visit the https://docs.quay.io/guides/create-repo.html#:~:text=via%20the%20UI-,To%20create%20a%20repository%20in%20the%20Quay.io%20UI%2C%20click,the%20'Create%20Repository'%20button.[quay documentation].
+
+Now we can create the container image for our WildFly project. Enter the following commands inside your Dockerfile: 
+```
+FROM quay.io/wildfly/wildfly-runtime-jdk11:latest
+COPY --chown=jboss:root dist/target/wildfly-29.0.0.Final-SNAPSHOT $JBOSS_HOME
+RUN chmod -R ug+rwX $JBOSS_HOME
+```
+Now run the following commands on your terminal from the WildFly directory to create an image and push it to your repository: 
+```
+docker build -t quay.io/<your-quay-username>/wildfly:wildfly-app .
+docker push quay.io/<your-quay-username>/wildfly:wildfly-app
+```
+The `-t` command indicates that we are specifying a tag, which in this case is `wildfly-app`. The tag is used to identify different images inside the same repository. Using this image, we can ensure that we are running your local version of WildFly, as opposed to the version of WildFly that is present on OpenShift. 
+
+== Deploying the Application 
```

Comment:
Should this be Building instead of deploying?

Reply:
changed to Build and deploy